src/otx/algo/hooks/recording_forward_hook.py

```diff
@@ -434,7 +434,10 @@ def average_and_normalize(
         Returns:
             np.array: Class-wise Saliency Maps. One saliency map per each class - [class_id, H, W]
         """
-        masks, scores, labels = (pred.masks, pred.scores, pred.labels)
+        if isinstance(pred, dict):
```

Comment:
`pred: InstanceData` at L425. However, why is the dictionary `pred` given here?

Reply:
Thank you, I updated description of the function